### PR TITLE
Fix Missing text when sorting from inside addon (Label2)

### PR DESCRIPTION
**Pull request by bigretromike**
## Description

This PR fix issue with `Label2` not populating properly inside `PluginDirectories` that where directories and not files.
It would be awesome if this is approved and backported to 18.x, so people can enjoy their addons with default skin :-)

## Motivation and Context

This PR resolve my origin issue: https://github.com/xbmc/xbmc/issues/14573 the way that @ronie wanted - in core and not in skin.

## How Has This Been Tested?

Compiled, tested with few video addons to check that its not a issue for anything else.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
@ronie imo this is a bug fix, not an improvement.

File: xbmc/filesystem/PluginDirectory.cpp [hunk 1]
```diff
@@ -277,92 +277,92 @@ void CPluginDirectory::AddSortMethod(int handle, SORT_METHOD sortMethod, const s
...
     case SORT_METHOD_SONG_RATING:
       {
-        dir->m_listItems->AddSortMethod(SortByRating, 563, LABEL_MASKS("%T", "%R"));
+        dir->m_listItems->AddSortMethod(SortByRating, 563, LABEL_MASKS("%T", "%R", "%T", "%R"));
         break;
       }
     case SORT_METHOD_YEAR:
       {
-        dir->m_listItems->AddSortMethod(SortByYear, 562, LABEL_MASKS("%T", "%Y"));
+        dir->m_listItems->AddSortMethod(SortByYear, 562, LABEL_MASKS("%T","%Y","%T","%Y"));
```
Comment (ksooo) [inline xbmc/filesystem/PluginDirectory.cpp]: Please add a blank after comma.

**Review by ksooo — COMMENTED** [review]
What was the criteria for changing some of the sort methods and not to adapt others? Why did you not extend all of the sort methods?

File: xbmc/filesystem/PluginDirectory.cpp (hunk 1 above)
Comment (bigretromike) [reply xbmc/filesystem/PluginDirectory.cpp]: fixed and pushed

**Comment by bigretromike** [comment]
@ksooo I wanted to fix issue I found https://github.com/xbmc/xbmc/issues/14573 - I added those that I was sure of, I wanted to fix the original issue with skin but @ronie point out that this should be fixed in core. And as this issue bothers me with k17, k18, k19 I wanted to fix the thing I was sure that need to be fix. If it was as simple as coping rest from somewhere I would do that too.

If you would like me to add more, please tell which one you want me to add.

**Comment by Rechi** [comment]
Please squash the commits.
Never add @<USER> in a commit message, as that person will receive an e-mail every time the commit is pushed somewhere on GitHub.

**Comment by bigretromike** [comment]
Sorry, about that, didn't know.
I will squash commits as soon as I read how to do that.

**Comment by bigretromike** [comment]
@Rechi I was able to squash them. Sorry for waiting.

**Comment by ksooo** [comment]
> I wanted to fix issue I found #14573 - I added those that I was sure of

 Fair enough.

**Comment by bigretromike** [comment]
I understand that jenkins fail is not anything with code `Caused: java.io.IOException: Backing channel 'JNLP4-connect connection from 10.0.2.2/10.0.2.2:49314' is disconnected.`

If this gonna be merge, could I ask for a backport ? I think that this is a bug fix (because the values were wrongs), also @ksooo stated the same and bug fix can have backport to 18.x. I'm hoping @ronie can change that.

If there anything I can assist more please let me know.

@ksooo I also added 2 more labels, nothing more I could came up with - I think that's more than enough for all those plugin developers.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Yep, this is a bug fix as currently it does not work as expected. Backport welcome.

**Review by ksooo — APPROVED** [review]
Thank you for your contribution. I will merge this once Jenkins is happy.

**Comment by bigretromike** [comment]
@ksooo can I do anything to make Jenkins happy ? 
Also should I push another PR to Leia as backport or that is something TeamKodi does ?

**Comment by ksooo** [comment]
> can I do anything to make Jenkins happy ?

 I don't think so. It's building right now.

> Also should I push another PR to Leia as backport

 Yes, please do.

**Comment by bigretromike** [comment]
@ksooo I added backport: https://github.com/xbmc/xbmc/pull/16517

**Pull request merged**